### ISSUE TITLE
Additional easy to use path for saving public data

### ISSUE THREAD

**Issue by bkmgit** [part-cw/lambdanative#455]
There are architecture specific paths to the file system for internal use by an application which are set in [system.c](https://github.com/part-cw/lambdanative/blob/master/libraries/liblambdanative/system.c).  For example on Android the following are defined:
- [system-dir](https://github.com/part-cw/lambdanative/blob/master/libraries/liblambdanative/system.c#L160)
- [system-appdir](https://github.com/part-cw/lambdanative/blob/master/libraries/liblambdanative/system.c#L161)
Android also offers [getExternalStoragePublicDirectory](https://developer.android.com/reference/android/os/Environment#getExternalStoragePublicDirectory(java.lang.String)) which was deprecated in API 29 but would seem to be helpful to have. For API 30, there is [shared datasets storage](https://developer.android.com/training/data-storage/shared/datasets) though probably JNI is needed to use this. 

Such a path may also be useful for [downloads](https://github.com/part-cw/lambdanative/blob/master/modules/download/download.scm#L46)

If using this is problematic, then implementing some way of allowing the user to search the file system in a GUI and specify where they would like to save a file or open a file would be helpful, but seems more difficult to implement in a cross-platform manner.

**Comment by mgorges** [comment]
Correct, as per 212ab34, we now use [getExternalFilesDirs](https://developer.android.com/reference/android/content/Context#getExternalFilesDirs(java.lang.String)) as `(system-directory)`, which is also accessible to outside tools; we also expose [getFilesDir](https://developer.android.com/reference/android/content/Context#getFilesDir()) as `(system-appdirectory)`. The change is not backwards compatible (i.e. required manual migration of files) as the previous location on `/sdcard` is no longer accessible to us so I know of people who disabled the patch and others who use it depending on what is more important to you - maintaining access to old files or supporting newer versions of Android. The content `(system-directory)` should be visible to file browsers and `adb` to you, though, unless we broke something recently.

Finally, sure, we can expose [getExternalStoragePublicDirectory](https://developer.android.com/reference/android/os/Environment#getExternalStoragePublicDirectory(java.lang.String)) easily if that is helpful, but I currently (and in the near future) do not have the capacity to build any new JNI-based file access pieces, so you'd have to submit a pull request for an implementation.

**Comment by bkmgit** [comment]
Ok, will make a pull request.
